signer/src/transaction_signer.rs

```diff
@@ -471,6 +484,23 @@ where
 
         self.send_message(BitcoinPreSignAck, &chain_tip.block_hash)
             .await?;
+
+        tracing::debug!(?chain_tip, "updating used_presign_blocks",);
+        self.used_presign_blocks.insert(*chain_tip);
```

Comment:
I think that we want to note that you've seen a message for this block immediately after checking whether we've seen something for this block, not after processing the message. A malicious coordinator could send junk that causes us to bail when we call `construct_package_sighashes`. Doing so would allow them to continually waste our compute as the signer processes their crappy request.